ui/src/app/utils/leo-runtime-initializer.tsx

```diff
@@ -204,7 +202,15 @@ export class LeoRuntimeInitializer {
       // - the user's most recent UserOverride config, if any
       // - (maybe) the user's most recently selected preset, if any
       // - general analysis
-      runtime = {...runtimePresets.generalAnalysis.runtimeTemplate};
+
+      if (this.currentRuntime && this.currentRuntime.status === RuntimeStatus.Deleted) {
```

Comment:
See TODO/comment above (and you can remove the TODO part). Ideally the logic here would be the same as you added on the create panel: if the deleted runtime has one of the preset configuration types, use that as the target (instead of whatever we got from the backend). This makes the behavior symmetric to opening the create panel, where we want to snap to the latest "preset", rather than whatever the preset happened to be when the user last used the system.

Note: this won't matter much for now as we don't expect the presets to change, but later when we start allowing configuration of docker image versions, this will be important so people don't unintentionally stuck on old versions.

Reply:
Right - made that change and added a test case